Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-roleserversideviewservice', function (Y) {
+    "use strict";
+    /**
+     * Provides the role server side view service class
+     *
+     * @method ez-roleserversideviewservice
+     */
+    Y.namespace('eZ');
+
+    /**
+     * The Role Server Side View Service class.
+     *
+     * @namespace eZ
+     * @class RoleServerSideViewService
+     * @constructor
+     * @extends eZ.ServerSideViewService
+     */
+    Y.eZ.RoleServerSideViewService = Y.Base.create('roleServerSideViewService', Y.eZ.ServerSideViewService, [], {
+        initializer: function () {
+            this.on('*:contentDiscover', function (e) {
+                e.config.contentDiscoveredHandler = Y.bind(this._assignRole, this);
+            });
+        },
+
+        /**
+         * Assign the role to the users/groups after the user has chosen them in
+         * the universal discovery widget
+         *
+         * @method _assignRole
+         * @protected
+         * @param {EventFacade} e
+         */
+        _assignRole: function (e) {
+            var data = e.target.get('data'),
+                userService = this.get('capi').getUserService(),
+                service = this;
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            userService.loadRole(data.roleId, function (error, response) {
+                var tasks = new Y.Parallel(),
+                    end = tasks.add(function (err, response) {
+                        if (err) {
+                            service._error('tasks.add() failed: ' + err.message);
```

Comment:
This message will be displayed to the user, and that's rather technical detail. This will be handled when `_assignRoleToEachAssignee` will fail, so maybe message that role wasn't assigned will be better.